libnetwork/drivers/overlay/overlay.go

```diff
@@ -184,10 +171,6 @@ func (d *driver) configure() error {
 	// Apply OS specific kernel configs if needed
 	d.initOS.Do(applyOStweaks)
 
-	if d.store == nil {
-		return nil
-	}
-
 	if d.vxlanIdm == nil {
 		return d.initializeVxlanIdm()
 	}
```

Comment:
When `d.store == nil` the function would return early, before reaching this block. Therefore this block should be unreachable when constant-propagating `(d.store == nil) == true`.

```suggestion
```

Reply:
Oh and that explains why I couldn't get how/when `d.vxlanIdm` was used. Good news is, we can completely throw out VNI allocation from this driver :slightly_smiling_face: